File: mahjong-main-private-api/src/main/java/com/github/mahjong/main/privateapi/dto/GameStartForm.java
package com.github.mahjong.main.privateapi.dto;

import com.github.mahjong.api.common.dto.WindDto;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class GameStartForm {

    /**
     * Players, that will participate in the game.
     */
    @NotNull
    @Size(min = 4, max = 4)
    public List<Long> players;
    /**
     * Players, that are seating on specific wind. If some player not specified their, it means it has random wind.
     */
    @NotNull
    @Size(max = 4)
    public Map<WindDto, Long> windToPlayer;
    @NotEmpty
    public String rulesSet;
    @NotEmpty
    public Set<WindDto> windsToPlay;
    public int startPoints;
    public boolean withUma;

}

File: mahjong-view/src/app/model/game.start.form.ts
import {Wind} from './wind';

export class GameStartForm {
    players: number[];
    windToPlayer: Map<Wind, number>;
    rulesSet: string;
    windsToPlay: Wind[];
    startPoints: number;
    withUma: boolean;

    static withSettings(rulesSet: string,
                        windsToPlay: Wind[],
                        startPoints: number,
                        withUma: boolean): GameStartForm {
        let form = new GameStartForm();
        form.rulesSet = rulesSet;
        form.windsToPlay = windsToPlay;
        form.startPoints = startPoints;
        form.withUma = withUma;
        return form;
    }
}